src/main/java/com/checkmarx/flow/service/SastScanner.java

```diff
@@ -71,10 +72,26 @@ public ScanResults scan(ScanRequest scanRequest) {
         checkScanSubmitEmailDelivery(scanRequest);
 
         try {
-            Integer projectId;
+            Integer scanId;
             CxScanParams cxScanParams = scanRequestConverter.toScanParams(scanRequest);
-            Integer scanId = cxService.createScan(cxScanParams, "CxFlow Automated Scan");
-            projectId = cxScanParams.getProjectId();
+            Integer projectId = cxScanParams.getProjectId();
+
+            log.info("Checking if there is any existing scan for Project:" + projectId);
+            Integer existingScanId = cxService.getScanIdOfExistingScanIfExists(projectId);
+
+            if (existingScanId != UNKNOWN_INT) {
+                if (flowProperties.getScanResubmit()) {
+                    log.info("Existing ongoing scan with id " + existingScanId +" found for Project :" + projectId);
```

Comment:
Template strings "{}" can be used in log messages for better readability.

Reply:
Replaced wherever applicable.